### DISCUSSION TITLE
Window Capture Disablement & Stack Protection Flag

### DISCUSSION THREAD

**Discussion by ghost** [sumatrapdfreader/sumatrapdf#4202] [Ideas]
I just thought SumatraPDF could benefit from Window Capture Disablement (also known as Streamer Mode or Screen Capture Block) feature to protect itself from host being able to take screenshots of content displayed in SumatraPDF and of SumatraPDF window itself. Also, I think "Stack Protection" flag should be added to SumtraPDF.exe image to enable full support for Kernel-Mode Hardware-Enforced Stack protection.

It would also be nice if SumatraPDF could perform some functions of QPDF, such as Linearization/Sanitation.

Many thanks for this awesome software!

**Comment by kjk** [comment]
Disable screenshots for what purpose? That was a rhetorical question, it would just make the app less useful.

I'm open to stack protection but not planning to work on that myself. Submit a patch: https://github.com/sumatrapdfreader/sumatrapdf/blob/master/premake5.lua

**Comment by GitHubRulesOK** [comment]
The one feature of SumatraPDF I use every day (many times most days) is to grab a screen snippet of an error condition so it would be pointless disabling a common reason for using a GUI in fact totally counter productive.

Acrobat Reader will totally screw up an existing file when it resaves as linearized. Since linearization is bloating. and thus slows down file loading it is designed simply to duplicate the tail in the head of a file to keep web transmission users happy. So has very little benefit for desktop users unless using a slow network viewer like Acrobat.

Sanitisation is not needed in SumatraPDF as its far more sanitised than MuPDF or Acrobat since it cannot run same exploitable ECMA code which MuPDF can.

**Comment by ghost** [comment]
There are situations when you stream video from your PC and need to see information on your screen that those who watch your stream cannot see. You see it. They don't.  System Informer uses "WDA_EXCLUDEFROMCAPTURE" to make that happen.

**Reply by GitHubRulesOK** [reply]
Sounds very niche use 
In similar situations I never had that problem with different types of private/company team viewers as
you show one desktop/area/screen (including SumatraPDF if needed) to others whilst keeping others like minefield progress or current private email info out of their viewport, thus often streaming only part of open applications even 1 of 2 full screen which could even be 2 SumatraPDF presentation screens to an audience.

One of the wider complaints about SumatraPDF is it is NOT screen accessible for use by the poor of sight so NVDA is a non runner in the documents.
